core/templates/pages/community-dashboard-page/translation-language-selector/translation-language-selector.component.ts

```diff
@@ -0,0 +1,114 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the translation language select.
+ */
+
+import {
+  Component, OnInit, Input, Output, EventEmitter, HostListener, ViewChild
+} from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+
+import { ContributionOpportunitiesBackendApiService } from
+  // eslint-disable-next-line max-len
+  'pages/community-dashboard-page/services/contribution-opportunities-backend-api.service';
+import { FeaturedTranslationLanguage } from
+  'domain/opportunity/FeaturedTranslationLanguageObjectFactory';
+import { LanguageUtilService } from 'domain/utilities/language-util.service';
+
+@Component({
+  selector: 'translation-language-selector',
+  templateUrl: './translation-language-selector.component.html'
+})
+export class TranslationLanguageSelectorComponent implements OnInit {
+  @Input() value: string;
+  @Output() setValue: EventEmitter<string> = new EventEmitter();
+  @ViewChild('dropdown', {'static': false}) dropdownRef;
+
+  options: {id: string, description: string}[];
+  languageIdToDescription: {[id: string]: string} = {};
+  featuredLanguages: FeaturedTranslationLanguage[] = [];
+
+  dropdownShown = false;
+  explanationPopupShown = false;
+  explanationPopupPxOffsetY = 0;
+  explanationPopupContent = '';
+
+  constructor(
+    private contributionOpportunitiesBackendApiService:
+      ContributionOpportunitiesBackendApiService,
+    private languageUtilService: LanguageUtilService
+  ) {}
+
+  ngOnInit() {
+    this.options = this.languageUtilService
+      .getAllVoiceoverLanguageCodes().map(languageCode => {
+        const description = this.languageUtilService
+          .getAudioLanguageDescription(languageCode);
+        this.languageIdToDescription[languageCode] = description;
+        return { id: languageCode, description };
+      });
+
+    this.contributionOpportunitiesBackendApiService
+      .fetchFeaturedTranslationLanguages()
+      .then((featuredLanguages: FeaturedTranslationLanguage[]) => {
+        this.featuredLanguages = featuredLanguages;
+      });
+  }
+
+  toggleDropdown() {
+    this.dropdownShown = !this.dropdownShown;
+  }
+
+  selectOption(value: string) {
+    this.setValue.emit(value);
+    this.dropdownShown = false;
+  }
+
+  showExplanationPopup(index: number) {
+    // Align popup to mouse-overed info icon
+    this.explanationPopupPxOffsetY = 75 + 30 * index;
+    this.explanationPopupContent = (
+      this.featuredLanguages[index].explanation);
+    this.explanationPopupShown = true;
+  }
+
+  hideExplanationPopup() {
+    this.explanationPopupShown = false;
+  }
+
+  /**
+   * Close popup when outside elements are clicked
+   * @param event mouse click event
+   */
+  @HostListener('document:click', ['$event'])
```

Comment:
Should we scope this event listener to small dom? Also, have you tried using tooltip or popover (see line 270 of editor-navigation.directive), can we do it that way?

Reply:
I believe the click listener must be attached to the document, correct me if I'm wrong. Also, I've looked into uib-tooltip, but I'm not sure that we have bootstrap installed for Angular2+, @marianazangrossi could you weigh in? 

Otherwise, the tooltip works fine, I believe.

Reply:
Hi @shavavo, I think you want to ask for @bansalnitish and @srijanreddy98, who are responsible for Angular 2+ features.

Reply:
Update: from email thread with Nitish and Srijan, adding bootstrap + material design is currently being investigated